pkg/scheduler/scheduler.go

```diff
@@ -475,10 +477,11 @@ func (s *Scheduler) forwardRequestToQuerier(querier schedulerpb.SchedulerForQuer
 	select {
 	case <-req.ctx.Done():
 		// If the upstream request is cancelled (eg. frontend issued CANCEL or closed connection),
-		// we need to cancel the downstream req. Only way we can do that is to close the stream (by returning error here).
+		// we need to cancel the downstream req. Only way we can do that is to return a gRPC error
+		// here with code Canceled and close the stream.
 		// Querier is expecting this semantics.
 		s.cancelledRequests.WithLabelValues(req.userID).Inc()
-		return req.ctx.Err()
+		return status.Error(codes.Canceled, context.Cause(req.ctx).Error())
```

Comment:
Note to reviewers: this change isn't required. The `TestCancelRequestInProgress_QuerierObservesCancellation` test added below passes with and without this change, as the gRPC library previously was translating the `context.Canceled` error to a gRPC-style error with code `Canceled`.

However, I believe this makes the intent clearer, and guards against unexpected future behaviour changes.